Demo/src/main/java/com/braintreepayments/demo/TransactionRequest.kt

```diff
@@ -0,0 +1,8 @@
+package com.braintreepayments.demo
+
+data class TransactionRequest(
+    val amount: String,
+    val paymentMethodNonce: String,
+    val merchantAccountId: String?,
+    val threeDSecureRequired: Boolean
```

Comment:
Take it or leave it, not sure if we want to group required vs optional together and flop the order here

Reply:
All the parameters are required to be passed into the constructor as there are no default values. Even though `merchantAccountId` is marked as nullable, an explicit `null` needs to be passed in.

I kept the ordering of the old Java constructor to keep the call sites the same:

```java
    public TransactionRequest(String amount, String paymentMethodNonce, String merchantAccountId, Boolean threeDSecureRequired) {
        ...
    }
```